### ISSUE TITLE
Control points

### ISSUE THREAD

**Issue by Kaffeine** [infclass/teeworlds-infclassR#67]
Consider some kind of capture (spawn?) points to help the infected <s>spread the word</s> spread the infection.

**Comment by Intramint** [comment]
Some elaboration: maps would have preset control points that can be captured by either team. By default, humans control all of them. capturing would probably mean staying in the zone for long enough without the enemy team interfering. A point controlled by the infected would allow them to spawn there after dying but only if there's no human already contesting it. This is to reduce the boring part of infected's gameplay which is walking from spawn to action. Control points would be placed somewhere along the line of defences. The infected would be able to choose where they want to spawn. Maybe the infection zones could also be divided into pseudo-capture points so the infected can choose where exactly to spawn especially on maps with big zombie spawns like newdust. A beneficial side effect to control point would be an encouragement for humans to retake space as the positive results would be much more visible. Currently a lot of people don't realize the importance of pushing for humans. All of this is however an overall buff to zombies. This system shouldn't replace the Witch as there would only be a few control points. Witches also can flank and can be harder to kill than it would be to claim a point.
Some uncertainties:
1. How would choosing a spawn position work?
2. What is the relation between spawning near a Witch and on a point? Do you choose to spawn on a Witch? What about the 50% chance?
3. What to do with the fact that the infected get a significant buff?

1: Creating an UI to choose the spawn might be hard but that may be the best option. Using chat commands isn't a good idea. An alternative is for the game to detect where the best place to spawn is but that has some disadvantages ie. if you want to hunt for roaming Heroes or if humans are camping in different places across the map. Letting a Witch choose the capture point wouldn't be great either because the Witch player might be new to the game and what happens if it dies?

2: The easiest solution would be to give the Witch a 100% spawn rate and treat it as one of the possible spawn points. However if it's kept at 50% then what happens? Maybe roll the 50% chance before you get to choose where to spawn? This way either you can pick the Witch spawn option or you can't and the game doesn't calculate anything when the spawn itself happens. The option to spawn on a Witch would be invisible or somehow marked as unavailable.

3: Just letting it be is an option since the teams are asymmetrical and the winrate isn't 50/50 anyway. An option is to somehow cripple the infected spawned in a zone e.g. by reducing their max HP to 10. Balancing it out with buffing individual human classes or nerfing infected classes doesn't seem like a good idea because this control point system doesn't change the early game where control points are on the humans' team. That would make the early game harder for zombies which might already be hard if there are inexperienced players in the initial infected team or on maps like eidalfitr where, in early game, hook aim alone isn't nearly enough to get a kill as all the humans are focused only in two spots. Another way is to create a sub-class/es for zombies that is/are given if spawned on a control point. They wouldn't be much of a threat by themselves but rather a pushing force which would be useful only when grouped with other zombies (sub-classes or normals). More experienced players can choose to spawn normally to make use of the class' abilities and because they can get back quicker thanks to their movement skills. Most of the less skilled players can't make use of their class' potential to the fullest anyway so going with a sub-class wouldn't be a major disadvantage. They are also much less likely to get a kill themself anyway since the better teammates are faster and more accurate. However they can feel like they're participating more often by having to do less walking. Most of such players probably won't even notice that they're weaker because they don't know the game anyway. We don't have to explicitly tell them that by choosing a spawn point they are at a disadvantage. That way they wouldn't feel as useless. This sub-class could also be given to anyone who spawns on a Witch and in turn give it a 100% chance to spawn instead of 50%. This would also solve question 2 but it's also meddling with the current game mechanics which require more testing. However a Witch nerf isn't bad considering that it's a late game infected class anyway. This means that it doesn't influence the aforementioned early game balance.

TL:DR: my opinion to solving all 3 problems: an UI to choose the spawn point. The Witch has a 100% spawning rate and is considered as one of the possible options when picking a spawn point. Spawning on a Witch or a control point gives you sub-class which has no special abilities and only 10 hp max (to reduce the reliance on the Engineer which would increase with control points). For now i think that they should insta-kill like other zombies but the hook damage could be removed.

**Comment by Kaffeine** [comment]
> 1. Creating an UI to choose the spawn

It is barely possible to create an extra UI without extending the protocol and customizing the client. Our human class selection menu already hitting the limits and highly interfere with other animations (the menu is implemented as a long animation).
This is only an option for the custom infclass client.

(I'll reply more a bit later)

**Comment by Intramint** [comment]
Maybe randomly choose where the infected spawns: either at the spawn with the normal class or on the Witch with the pusher sub-class. If there is no Witch then spawn at the control point with the same sub-class. If you selfkill during the first 3 seconds of spawning, reduce the respawn time and guarantee a spawn at the location you haven't spawned in. For example, spawn in the normal infection zone -> selfkill within 3 seconds -> wait much less for a respawn -> spawn near a Witch/control point
